.github/workflows/0-on-demand.yaml

```diff
@@ -0,0 +1,94 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+name: On Demand Tests
+
+on:
+  workflow_dispatch:
+    inputs:
+      enable_ngrok_debug:
+        description: "Enable Ngrok Debugging"
+        required: true
+        type: boolean
+        default: false
+      test_suite:
+        description: Select Test Suite to run
+        type: choice
+        options:
+        - Unit
+        - System
+        - MultiRuntime
+        - Standalone
+        - Scheduler
+        - Performance
+        ## Uncomment those to debug the build
+        - Dummy
+        #- DummyKO
+
+env:
+  # openwhisk env
+  TEST_SUITE: ${{ inputs.test_suite }}
+  ANSIBLE_CMD: "ansible-playbook -i environments/local -e docker_image_prefix=testing"
+  GRADLE_PROJS_SKIP: ""
+
+  # (optional) slack incoming wehbook for notifications
+  SLACK_WEBHOOK: ${{secrets.SLACK_WEBHOOK}}
+
+  # (optional) s3 log upload
+  AWS_BUCKET: ${{ secrets.AWS_BUCKET }}
```

Comment:
Same for NGROK, so now contributors are in charge of uploading logs to their own environment to figure out the failed tests of a PR.
For periodic tests against the upstream master branch, those who have permission can configure proper secrets for the upstream repo.

It needs to be discussed where to place such logs.

Reply:
Yes you can put your own AWS_BUCKET where you want. The bucket can be made public as I did here: https://nuvolaris-logs.s3.eu-central-1.amazonaws.com/index.html.
Instructions are in the README.
Actually, since there is already a bucket configured for some tests, it will be used!
Maybe this is not what the owner of the bucket wants.
We need to get a bucket from Amazon or from some other cloud provider, I am sure there are donations somewhere of resources that we can use...